framework/resource/retryresource/crud_resource.go

```diff
@@ -0,0 +1,230 @@
+package retryresource
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "retry"
```

Comment:
IMO this name should reflect the type. Now we have the same name for different resources. We should not do this. `basicresource` and `crudresource` works for me.